GH1734: Add GitLink 3 aliases

Cheesebaron opened the pull request:
This Pull Request relates to https://github.com/cake-build/cake/issues/1734 and updates the GitLink tool to reflect the major restructure that was done in GitLink 3.0.0.

Instead of taking a solution directory, the new GitLink tool takes singular pdb files and patches them. I've added a helper alias to allow passing a FilePathCollection to allow globbing.

dnfclas commented:
@Cheesebaron,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

patriksvensson commented:
This will be a breaking change for anyone not using GitLink 3.0, right? Any possibility to introduce a GitLink3-runner or similar?

Cheesebaron commented:
@patriksvensson correct. This could be made as a different runner and alias. I have no issue with doing that instead.

@devlead will fix that.

devlead commented:
Agree with @patriksvensson, could you keep old aliases and add new 3 editions

Cheesebaron commented:
Should the AliasCategory still be GitLink or a separate GitLink3 one instead?

devlead commented:
![smartselectimage_2017-08-07-20-25-22](https://user-images.githubusercontent.com/1647294/29040264-a2fcf9e4-7bae-11e7-8922-07a0411bdb6d.png) do seperate like NUnit

Cheesebaron commented:
Done. Old Gitlink stuff is as is. I've added version pinning in the XML docs for the v2 stuff.

I've renamed all the new GitLink stuff to GitLink3 and made a category called GitLink v3, similar to NUnit.

mholo65 commented on src/Cake.Common/Tools/GitLink/GitLink3Aliases.cs:
```diff
@@ -0,0 +1,114 @@
+    /// <para>
+    /// In order to use the commands for this alias, include the following in your build.cake file to download and
+    /// install from NuGet.org, or specify the ToolPath within the <see cref="GitLink3Settings" /> class:
+    /// <code>
```
Malformed Xml. Remove the extra `<para>` above.

mholo65 replied:
https://ci.appveyor.com/project/cakebuild/cake/build/0.22.0-PullRequest.1735+58.build.3758#L553

Cheesebaron replied:
@mholo65 sooooo I just copied what was in: https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/GitLink/GitLinkAliases.cs#L13 it is the same...

mholo65 replied:
I see, it seems to be complaining about `version` 

`Tools\GitLink\GitLinkAliases.cs(18,38): error CS1570: XML comment has badly formed XML -- 'version'`

On second glimpse, to me that XML looks fine.

Cheesebaron replied:
I've fixed it. The ampersand needed escaping

Cheesebaron commented:
@devlead & @mholo65 anything else you guys want me to change? I think I have addressed both the issues you pointed out. I've rebased the PR so many times now, I can rebase it again when you think it is ready to merge.

devlead commented:
@Cheesebaron your changes have been merged, thanks for your contribution 👍
